### DISCUSSION TITLE
The matching issue between the ExtPtfm module in OpenFAST v2.3.0 and the ExtPtfm.dat file and ExtPtfm_SE.dat file in the current r-test repository of OpenFAST on GitHub.

### DISCUSSION THREAD

**Discussion by Steven647** [OpenFAST/openfast#1975] [Q&A] [answered]
Dear All,

First of all, I would like to express my sincere appreciation to the developers of OpenFAST for their continuous contributions to this software and the positive atmosphere in the GitHub discussions.

Recently, I have been attempting to **prescribe the motion** of 5MW floating offshore wind turbine in OpenFAST V2.3.0. As mentioned in the Issue #617  titled "Develop Prescribed Platform Motion capability with aero-elasticity", I aim to achieve the prescribed motion of the turbine by applying loads through **ExtPtfm** module. 

I have used the example _ExtPtfm.dat_ and _ExtPtfm_SE.da_t files provided in the **current** **r-test repository** of OpenFAST. However, when running OpenFAST, I encountered the error shown in the image below.

![批注 2024-01-12 200752](https://github.com/OpenFAST/openfast/assets/71825869/a7ba06fc-82ac-4ed8-89a7-0be939b4667b)

Therefore, I would like to inquire whether the reason for the error is a **mismatch** between the versions of **OpenFAST v2.3.0** and **the two input files used for ExtPtfm module**. If the error is indeed due to the version mismatch, and considering that I am currently coupling OpenFAST v2.3.0 with SOWFA, which is a wind farm wake simulation software, is it possible to modify the input files of the ExtPtfm module **without** changing the version of OpenFAST to make OpenFAST successfully invoke the ExtPtfm module and achieve the prescribed turbine motion?

Looking forward to your answers!

**Comment by jjonkman** [comment]
Dear @Steven647,

Can you clarify where you obtained the _ExtPtfm.dat_ and _ExtPtfm_SE.dat_ files you are using from?  Are you referring to those available from the readthedocs: https://openfast.readthedocs.io/en/main/source/user/extptfm/appendix.html?

Best regards,

**Reply by Steven647** [reply]
Dear @jjonkman ,

Thank you very much for your reply. 

As you said, the _ExtPtfm.dat_ and _ExtPtfm_SE.dat_ files that I am using 
are from the readthedocs: https://openfast.readthedocs.io/en/main/source/user/extptfm/appendix.html.

Do you think this is an error caused by the **version mismach**? Furthermore, is it possible to modify the input files of the ExtPtfm module **without** changing the version of OpenFAST to make OpenFAST successfully invoke the ExtPtfm module and achieve the prescribed motion of wind turbine?

Best Regards,

**Reply by jjonkman** [reply, marked as answer]
Dear @Steven647,

According to pull request history, the current version of the `ExtPtfm` module that relies on the input files you are using was introduced in OpenFAST v2.4.0 via the following PR: https://github.com/OpenFAST/openfast/pull/344.  To use this version of `ExtPtfm` within OpenFAST v2.3.0 will require that you merge this PR into the v2.3.0 code base and recompile.

Best regards,
